repository_service_tuf_worker/repository.py

```diff
@@ -287,87 +308,89 @@ def _update_snapshot(
-        """
-        logging.debug(f"Adding to unpublished meta {targets_meta}")
-        if len(targets_meta) == 0:
-            logging.info("Nothing to send to be published")
-            return None
-
-        with self._redis.lock("TUF_TARGETS_META"):
-            if self._redis.exists("unpublished_metas"):
-                targets_waiting_commmit = self._redis.get(
-                    "unpublished_metas"
-                ).decode("utf-8")
-                for bins_name, _ in targets_meta:
-                    if bins_name not in targets_waiting_commmit:
-                        self._redis.append(
-                            "unpublished_metas", f", {bins_name}"
-                        )
-            else:
-                self._redis.set(
-                    "unpublished_metas",
-                    ", ".join(bins_name for bins_name, _ in targets_meta),
-                )
-
-    def _publish_meta_state(
-        self, targets_meta: List[Tuple[str, int]], update_state: Optional[str]
-    ) -> List[Optional[Tuple[str, int]]]:
+    def _update_task(
+        self,
+        bin_targets: Dict[str, List[targets_models.RSTUFTargets]],
+        update_state: Task.update_state,
+    ):
         """
         Updates the 'RUNNING' state with details if the meta still not
         published in the latest Snapshot. It runs every 3 seconds.
         """
-        logging.debug(f"waiting metas to be published {targets_meta}")
+        logging.debug(f"Waiting roles to be published {list(bin_targets)}")
 
-        def _update_state(targets_meta: List[Tuple[str, int]]):
-            unpublised_roles = [
-                f"{role} version {version}" for role, version in targets_meta
-            ]
+        def _update_state(
+            bin_targets: Dict[str, List[targets_models.RSTUFTargets]],
+            completed_roles: List[str],
+        ):
             update_state(
                 state="RUNNING",
                 meta={
-                    "unpublished_roles": unpublised_roles,
+                    "published_roles": completed_roles,
+                    "roles_to_publish": f"{list(bin_targets.keys())}",
                     "status": "Publishing",
                     "last_update": datetime.now(),
                 },
             )
 
         while True:
-            snapshot = self._load(Roles.SNAPSHOT.value)
-            for target_meta in targets_meta:
-                snapshot_meta_file = snapshot.signed.meta[
-                    f"{target_meta[0]}.json"
-                ]
-                if snapshot_meta_file.version == target_meta[1]:
-                    logging.debug(f"Found published meta {target_meta}")
-                    targets_meta.remove(target_meta)
-
-            if len(targets_meta) > 0:
-                _update_state(targets_meta)
+            completed_roles: List[str] = []
+            for role_name, targets in bin_targets.items():
+                for target in targets:
+                    self._db.refresh(target)
```

Comment:
What exactly does this do?
Does it update the database with this new `target` information or what?

Reply:
The SQLAlchemy DB Session updates the target object that another thread in the SQL might update.

Reply:
It refreshes from the database, to read the latest committed data.